csharp/src/Apache.Arrow/Field.Builder.cs

```diff
@@ -70,6 +70,19 @@ public Builder Metadata(string key, string value)
                 return this;
             }
 
+            public Builder Metadata(IEnumerable<KeyValuePair<string, string>> dict)
+            {
+                foreach (KeyValuePair<string, string> entry in dict)
+                {
+                    if (string.IsNullOrWhiteSpace(entry.Key))
```

Comment:
Can you just call the `Metadata` overload above in here? That way you don't need to duplicate the implementation.